android/2_build_toolchain.sh

```diff
@@ -5,108 +5,62 @@ set -e
 
 export WORKSPACE=$PWD
 
-export NDK_ROOT=$WORKSPACE/android-ndk-r15c
-export SDK_ROOT=$WORKSPACE/android-sdk
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source $SCRIPT_DIR/../shared/import
 
 # Number of CPU
-NBPROC=$(getconf _NPROCESSORS_ONLN)
-
-# Setup PATH
-PATH=$PATH:$NDK_ROOT:$SDK_ROOT/tools
+nproc=$(nproc)
 
 # Use ccache?
-if [ -z ${NO_CCACHE+x} ]; then
-	if hash ccache >/dev/null 2>&1; then
-		ENABLE_CCACHE=1
-		echo "CCACHE enabled"
-	fi
-fi
+test_ccache
 
 if [ ! -f .patches-applied ]; then
 	echo "patching libraries"
 
+	patches_common
+
+	cp -rup icu icu-native
+
 	# Accept api9 for make_standalone_toolchain
 	perl -pi -e 's/min_api = 14/min_api = 9/' ./android-ndk-r15c/build/tools/make_standalone_toolchain.py
 
 	# Patch cpufeatures, hangs in Android 4.0.3
 	patch -Np0 < cpufeatures.patch
 
-	# disable pixman examples and tests
-	cd pixman-0.34.0
-	perl -pi -e 's/SUBDIRS = pixman demos test/SUBDIRS = pixman/' Makefile.am
-	autoreconf -fi
-	cd ..
-
 	# disable unsupported compiler flags by clang in libvorbis
-	cd libvorbis-1.3.5
-	perl -pi -e 's/-mno-ieee-fp//' configure
-	cd ..
-
-	# update autoconf stuff and preprocessor macro to recognize android
-	cd libxmp-lite-4.4.1
-	patch -Np1 < ../libxmp-a0288352.patch
-	cd ..
-
-	# disable libsndfile examples and tests
-	cd libsndfile-1.0.28
-	perl -pi -e 's/ examples regtest tests programs//' Makefile.am
-	autoreconf -fi
-	cd ..
+	perl -pi -e 's/-mno-ieee-fp//' $LIBVORBIS_DIR/configure
 
-	# Wildmidi: Disable libm
-	cd wildmidi-wildmidi-0.4.2
-	perl -pi -e 's/FIND_LIBRARY\(M_LIBRARY m REQUIRED\)//' CMakeLists.txt
-	cd ..
+        # Wildmidi: Support install for CMAKE_SYSTEM_NAME Generic
+        pushd $WILDMIDI_DIR
+        patch -Np1 < $SCRIPT_DIR/../shared/extra/wildmidi-generic-install.patch
+        popd
 
 	# use android config
-	cd SDL2-2.0.6
+	pushd $SDL2_DIR
 	mv include/SDL_config_android.h include/SDL_config.h
 	mkdir -p jni
-	cd ..
+	popd
 
 	touch .patches-applied
 fi
 
-# generic autotools library installer
-function install_lib {
-	cd $1
-	shift
-	./configure --host=$TARGET_HOST --prefix=$PLATFORM_PREFIX --disable-shared --enable-static $@
-	make clean
-	make -j$NBPROC
-	make install
-	cd ..
-}
-
-# generic cmake library installer
-function install_lib_cmake {
-	cd $1
-	shift
-	rm -rf CMakeCache.txt CMakeFiles/
-	cmake . -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-		-DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=$PLATFORM_PREFIX $@
-	make clean
-	make -j$NBPROC
-	make install
-	cd ..
-}
-
 # Install mpg123
 function install_lib_mpg123 {
 	export CPPFLAGSOLD=$CPPFLAGS
 	export CPPFLAGS="$CPPFLAGS -DHAVE_MMAP"
-	install_lib mpg123-1.25.8 --enable-fifo=no --enable-ipv6=no --enable-network=no \
-		--enable-int-quality=no --with-cpu=generic --with-default-audio=dummy
+	rm $PLATFORM_PREFIX/config.cache
+	install_lib $MPG123_DIR $MPG123_ARGS
+	rm $PLATFORM_PREFIX/config.cache
```

Comment:
This totally breaks the purpose why we use a build system cache!
I would like to investigate why `mmap` is not detected/used instead.

Reply:
good point, I will recompile to check if this was fixed in the meanwhile.